### DISCUSSION TITLE
Still can't use planetscale due to foreign keys

### DISCUSSION THREAD

**Discussion by joeljackson** [prisma/prisma#10245] [Q&A] [answered]
Giving this a shot with the `referentialIntegrity=prisma` flag

Migration file

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  shadowDatabaseUrl    = env("SHADOW_DATABASE_URL")
  referentialIntegrity = "prisma"
}

model User {
  id           Int      @id @default(autoincrement())
  createdAt    DateTime @default(now())
  updatedAt    DateTime @updatedAt
  pii_document String   @db.VarChar(255)
  Test         Test[]
}

model Test {
  id         Int      @id @default(autoincrement())
  createdAt  DateTime @default(now())
  updatedAt  DateTime @updatedAt
  identifier String   @db.VarChar(255)
  user       User     @relation(fields: [userId], references: [id])
  userId     Int

  @@index([userId])
}
```

Then getting
```
Error: P3006

Migration `20211113180521_add_tests` failed to apply cleanly to the shadow database. 
Error code: P3018
Error:
A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 20211113180521_add_tests

Database error code: 1317

Database error:
Foreign keys cannot be created on this database. Learn more how to handle this: https://pris.ly/d/migrate-no-foreign-keys

Please check the query number 2 from the migration file.
```
Details
```
"@prisma/client": "^3.4.2"

**Comment by nboukeffa** [comment, marked as answer]
You probably already have a failed migration. After `rm -rf prisma/migrations` and `npx prisma migrate dev --name init` it worked well for me.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
